packages/volto/src/components/manage/Controlpanels/Users/RenderUsers.jsx

```diff
@@ -13,199 +14,174 @@ import Toast from '@plone/volto/components/manage/Toast/Toast';
- * @extends Component
+ * RenderUsers functional component.
+ * @function RenderUsers
  */
-class RenderUsers extends Component {
-  /**
-   * Property types.
-   * @property {Object} propTypes Property types.
-   * @static
-   */
-  static propTypes = {
-    user: PropTypes.shape({
-      username: PropTypes.string,
-      fullname: PropTypes.string,
-      roles: PropTypes.arrayOf(PropTypes.string),
-    }).isRequired,
-    roles: PropTypes.arrayOf(
-      PropTypes.shape({
-        id: PropTypes.string,
-      }),
-    ).isRequired,
-    onDelete: PropTypes.func.isRequired,
-    isUserManager: PropTypes.bool.isRequired,
-  };
+const RenderUsers = (props) => {
+  const [user, setUser] = useState({});
+  const prevUpdateRequestRef = useRef();
 
-  /**
-   * Constructor
-   * @method constructor
-   * @param {Object} props Component properties
-   * @constructs Sharing
-   */
-  constructor(props) {
-    super(props);
-    this.state = {
-      user: {},
-    };
-    this.onChange = this.onChange.bind(this);
-    this.onEditUserError = this.onEditUserError.bind(this);
-    this.onEditUserSubmit = this.onEditUserSubmit.bind(this);
-  }
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const updateRequest = useSelector((state) => state.users?.update);
+
+  const {
+    user: propsUser,
+    listUsers,
+    updateUser: updateUserRole,
+    isUserManager,
+    roles,
+    inheritedRole,
+    userschema,
+    onDelete,
+  } = props;
+
+  // Use dispatch to call updateUser action
+  const updateUserData = (userId, userData) => {
+    dispatch(updateUser(userId, userData));
+  };
 
   /**
    * @param {*} event
    * @param {*} { value }
-   * @memberof UsersControlpanelUser
    */
+  const onChange = (_event, { value }) => {
+    const [userId, role] = value.split('&role=');
+    updateUserRole(userId, role);
+  };
 
-  onChange(event, { value }) {
-    const [user, role] = value.split('&role=');
-    this.props.updateUser(user, role);
-  }
+  // Handle successful user update
+  useEffect(() => {
+    const prevUpdateRequest = prevUpdateRequestRef.current;
 
-  componentDidUpdate(prevProps) {
+    // Only execute if there was a transition from loading to loaded
     if (
-      prevProps.updateRequest.loading &&
-      this.props.updateRequest.loaded &&
-      this.state?.user?.id === this.props?.user?.id
+      prevUpdateRequest?.loading &&
+      updateRequest?.loaded &&
+      user?.id === propsUser?.id
     ) {
-      this.setState({ user: {} });
-      this.props.listUsers();
-      return toast.success(
+      setUser({});
+      if (listUsers) {
+        listUsers();
+      }
+      toast.success(
         <Toast
           success
-          title={this.props.intl.formatMessage(messages.success)}
-          content={this.props.intl.formatMessage(messages.updateUserSuccess)}
+          title={intl.formatMessage(messages.success)}
+          content={intl.formatMessage(messages.updateUserSuccess)}
         />,
       );
     }
```

Comment:
I find it simpler to use the promise returned from dispatching an action. e.g.:
```jsx
dispatch(updateUser(userId, userData))
  .then((result) => {
    // handle success here
  }).catch((error) => {
    // handle error here
  });
```

Then there's no need for keeping track of the previous updateRequest just to detect when it finished.